torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,290 @@
+        for fname in glob.iglob(pathname, recursive=True):
+            if not fname.endswith(".tar.gz"):
+                return
+
+        # Check if the tar files have already been downloaded
+        pathname = os.path.join(self.root, "*.tar.gz")
+        if glob.glob(pathname):
+            self._extract()
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        for biome, md5 in self.md5s.items():
+            download_url(
+                self.url.format(biome), self.root, md5=md5 if self.checksum else None
+            )
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        pathname = os.path.join(self.root, "*.tar.gz")
+        for tarfile in glob.iglob(pathname):
+            extract_archive(tarfile)
+
+    def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
+        """Retrieve image/mask and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            sample of image, mask and metadata at that index
+
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        hits = self.index.intersection(tuple(query), objects=True)
+        img_filepaths = cast(List[str], [hit.object for hit in hits])
+
+        if not img_filepaths:
+            raise IndexError(
+                f"query: {query} not found in index with bounds: {self.bounds}"
+            )
+
+        data_list: List[Tensor] = []
+
+        for band in self.all_bands:
+            band_filepaths = []
+            for img_filepath in img_filepaths:
+                if band in img_filepath:
+                    filepath = img_filepath
+                    filename = os.path.basename(filepath)
+                    directory = os.path.dirname(filepath)
+                    match = re.match(self.filename_regex, filename)
+                    if match:
+                        if "date" in match.groupdict():
+                            start = match.start("band")
+                            end = match.end("band")
+                            filename = filename[: start - 1] + band + filename[end:]
+                    filepath = os.path.join(directory, filename)
+                    band_filepaths.append(filepath)
+                    data_list.append(self._merge_files(band_filepaths, query))
+
+        img = torch.cat(data_list)
+
+        mask_filepaths = [
```

Comment:
This is super hard to read. Can you convert it from a list comprehension to a for-loop?

Reply:
Yes, I will convert it to a for-loop for easier understanding. Thank you.